File: Scripts/volatile_and_force_compiler_task/classes/c-sharp/classes.cs
using System;

class MyClass{

  // instance variable
  private volatile int variable;  // Note: instance variables are usually "private"

  /**
  * The constructor
  */
  public MyClass(){
    // creates a new instance
  }

  /**
  * A method
  */
  public void someMethod(int i){
   this.variable = 1 + i;
  }

  private static volatile int r = 1;

  public static int executeTask(int i) {

	MyClass obj = new MyClass();
	obj.someMethod(i);

	return i;
  }

public static void Main(String[] args){

	for (int i = 0; i < 2100000000; ++i) {
		r = executeTask(i + r);
	}
}

}

File: scripts/task_tested_with_compile_script/classes/javascript/classes.js
class MyClass{
	constructor(variable) {
		this.variable = variable;
	}

	someMethod(i) {
		this.variable = this.variable + i;
	}
}

function executeTask(i) {
	const obj = new MyClass(123456);
	obj.someMethod();
	return 1;
}

var r = 1;

for (var i = 0; i < 2100000000; ++i) {
	r = executeTask(i + r);
}
